### PR TITLE
Transaction detail

### PR DESCRIPTION
**Pull request by Abhityagi16**
- Transaction Detail UI
-  Added click listener in transaction list

### CODE REVIEW

File: app/src/main/java/com/tari/android/wallet/ui/activity/transactiondetails/TransactionDetailActivity.kt [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            return@setOnKeyListener false
+        }
+
+        val calendar = Calendar.getInstance()
+        calendar.timeInMillis = transaction!!.timestamp.toLong() * 1000
+        formatDate(calendar)
+
+    }
+
+    private fun formatDate(cal: Calendar) {
```
Comment (kukabi) [inline app/src/main/java/com/tari/android/wallet/ui/activity/transactiondetails/TransactionDetailActivity.kt]: Did you try SimpleDateFormat or something like it? This function is redundant if you can format it with a formatter - please refer to other files for examples.

File: app/src/main/res/layout/activity_transaction_detail.xml [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            android:textColor="@color/black"
+            android:textStyle="bold"
+            app:customFont="AVENIR_LT_STD_HEAVY"
+            app:layout_constraintBottom_toBottomOf="@id/tx_detail_back"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@id/tx_detail_back" />
+
+        <com.tari.android.wallet.ui.component.CustomFontTextView
+            android:id="@+id/tx_detail_date"
```
Comment (kukabi) [inline app/src/main/res/layout/activity_transaction_detail.xml]: I think I left a comment about the id literals.

File: app/src/main/res/layout/activity_transaction_detail.xml [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_alignParentEnd="true"
+                    android:layout_marginEnd="24dp"
+                    android:text="@string/add_contact"
+                    android:textColor="@color/home_bg_gradient_center"
+                    app:customFont="AVENIR_LT_STD_MEDIUM" />
+            </RelativeLayout>
+
+
```
Comment (kukabi) [inline app/src/main/res/layout/activity_transaction_detail.xml]: Please don't use double empty lines if you're not emphasizing something - and in that case you can use XML comments.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -94,4 +94,18 @@
     <!-- debug log - title -->
     <string name="debug_log_title">Log</string>
 
+<!--    Transaction Details-->
+    <string name="payment_received">Payment Received</string>
```
Comment (kukabi) [inline app/src/main/res/values/strings.xml]: Please id your string resources just like the way you id your layout components.

[activity_or_fragment_name]_[string_resource_name]

For this instance it it tx_details_payment_received.

If you think the resource is something more general and can be used by other views, then you can move it around the top of the file and prefix it with common, such as:

common_payment_received

**Comment by CjS77** [comment]
Covered in #114

**Pull request closed without merging**